src/model.rs

```diff
@@ -144,14 +144,21 @@ impl ProofData {
         }
     }
 }
-#[derive(Serialize, Deserialize, Default, PartialEq, Eq, Clone, Encode, Decode, Hash)]
+#[derive(Serialize, Deserialize, Default, PartialEq, Eq, Clone, Encode, Decode)]
 pub struct ProofTransaction {
     // TODO: investigate if we can remove blob_tx_hash. It can be reconstrustruced from HyleOutput attributes (blob + identity)
     pub blob_tx_hash: TxHash,
     pub proof: ProofData,
     pub contract_name: ContractName,
 }
 
+#[derive(Serialize, Deserialize, Default, PartialEq, Eq, Clone, Encode, Decode)]
+pub struct RecursiveProofTransaction {
+    pub via: ContractName,
+    pub proof: ProofData,
+    pub verifies: Vec<(TxHash, ContractName)>,
```

Comment:
Given a BlobTx, If there is multiple blobs for the same contract, we will have twice the same tuple right ? It works the same way as for "ProofTransaction" ?

Reply:
yup